services/clients/firebase_client.py

```diff
@@ -9,48 +9,74 @@
 from google.cloud.firestore_v1.client import Client
 from google.cloud.storage import Blob
 
+
 class FirebaseClient:
-    TEST_COLLECTION = "documentation"
+    DOCUMENTATION_COLLECTION = "documentation"
 
     def __init__(self):
         self.bucket = storage.bucket()
         self.db: Client = firestore.client()
 
-    def add_document(self, collection_name, data) -> DocumentReference:
-        collection_ref = self.db.collection(collection_name)
+    def add_blob(self, blob_url, data):
+        blob = self.bucket.blob(blob_url)
+        blob.upload_from_string(data)
+
+    def get_blob(self, blob_url) -> Blob:
+        return self.bucket.get_blob(blob_url)
+
+    def get_documentation(self, doc_id) -> Dict[str, Any] | None:
+        document_snapshot = self._get(self.DOCUMENTATION_COLLECTION, doc_id)
+        if not document_snapshot:
+            return None
+
+        document_dict = document_snapshot.to_dict()
+        document_dict["id"] = document_snapshot.id
+
+        return document_dict
+
+    def add_documentation(self, data) -> str:
```

Comment:
these functions are now specific to our use-cases. Which all deal with the `DOCUMENTATION_COLLECTION`. That way no one outside of `FirebaseClient` has to understand/know what our collection path is or how we are using it.

I also made it return the `id` itself, instead of the `document_ref` because I realized we only ever use the `id` from the `document_ref` and don't really need it. This is another way of encapsulating the Firestore logic from other classes or parts of the app.